src/main/java/dino/command/AddTaskCommand.java

```diff
@@ -0,0 +1,123 @@
+     */
+    public AddTaskCommand(String taskString, Parser.CMDTYPE taskType) {
+        this.taskString = taskString;
+        this.taskType = taskType;
+    }
+
+    /**
+     * Executes the command to add the specific type of task to the task list
+     *
+     * @param storage the local storage file
+     * @param taskList the current task list to which the new task will be added
+     * @return the output message after execution
+     * @throws TimeNotSpecifiedException if the task is of type Deadline or Event
+     * but the time is not specified
+     * @throws EmptyTaskDescriptionException if the description of the task is empty
+     * @throws InvalidFormatException if the task is of type Deadline or Event
+     * but the time is not entered in the format of "yyyy-mm-dd"
+     */
+    @Override
+    public String execute(Storage storage, TaskList taskList) throws
+            TimeNotSpecifiedException, EmptyTaskDescriptionException, InvalidFormatException {
+        String description = getTaskDescription(this.taskString, this.taskType);
+        Task task = null;
+        switch (taskType) {
+        case TODO: {
+            task = new ToDo(description);
+            break;
+        }
+        case DEADLINE: {
+            LocalDate time = getTaskTime(this.taskString);
+            task = new Deadline(description, time);
+            break;
+        }
+        case EVENT : {
+            LocalDate time = getTaskTime(this.taskString);
+            task = new Event(description, time);
+            break;
+        }
+        default: {
+            break;
+        }
+        }
+        return taskList.addTask(task);
+    }
+
+    /**
+     * Extracts the description of the task from a user input command to add task
+     *
+     * @param s the user input command
+     * @param type the type of the task
+     * @return the description of the task that is extracted from the user input command
+     * @throws EmptyTaskDescriptionException if the description of the task is empty
+     * @throws TimeNotSpecifiedException if the task is of type Deadline or Event but the time is not specified
+     */
+    public static String getTaskDescription(String s, Parser.CMDTYPE type) throws EmptyTaskDescriptionException, TimeNotSpecifiedException {
+        if (s.length() < type.toString().length() + 2) {
+            throw new EmptyTaskDescriptionException();
+        }
+        switch (type) {
+        case TODO: {
+            return s.substring(5).trim();
+        }
+        case DEADLINE:
+        case EVENT: {
+            if (s.contains("/by ") || s.contains("/at ")) {
+                return s.substring(type.toString().length() + 1, s.indexOf("/")).trim();
+            } else {
+                throw new TimeNotSpecifiedException(type.toString());
+            }
+        }
+        default: {
+            break;
+        }
+        }
+        return s;
```

Comment:
Hmm do you think it's better to just move the return into the default case and remove the break statement? I feel that it would make the code look cleaner! Also, the curly braces could be removed as well in each of the cases.

```suggestion
        default: 
            return s;
        }
```